contracts/cw-names-registry/src/error.rs

```diff
@@ -0,0 +1,26 @@
+use cosmwasm_std::StdError;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ContractError {
+    #[error("{0}")]
+    Std(#[from] StdError),
+
+    #[error("Unauthorized")]
+    Unauthorized {},
+
+    #[error("Insufficient funds sent")]
+    InsufficientFunds {},
+
+    #[error("This name is reserved for later user")]
+    NameReserved {},
+
+    #[error("This name is not reserved for later use")]
+    NameNotReserved {},
+
+    #[error("This name is already taken")]
+    NameAlreadyTaken {},
+
+    #[error("You already registered a name with this DAO")]
+    NameAlreadyRegistered {},
```

Comment:
`NameAlreadyRegistered` makes it sound like the name is already taken. Perhaps we could name this one `AddressAlreadyHasOneName`? It's cumbersome, but boneheadedly obvious. 


Some alternatives, in the interest of throwing spaghetti at the wall:

- `AlreadyRegisteredOneName`
- `AlreadyRegisteredName`
- `AlreadyHasName`